tests/test_runner.py

```diff
@@ -0,0 +1,166 @@
+# material is made available to anyone wishing to use, modify, copy, or
+# redistribute it subject to the terms and conditions of the GNU General Public
+# License v.2 or later.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+import os
+import tempfile
+import unittest
+
+import xml.etree.ElementTree as etree
+import mock
+
+from skt import runner
+
+SCRIPT_PATH = os.path.dirname(__file__)
+
+DEFAULT_ARGS = {
+    'jobtemplate': '{}/assets/test.xml'.format(SCRIPT_PATH)
+}
+
+
+class TestRunner(unittest.TestCase):
+    """Test cases for runner module"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up test fixtures"""
+        with open('{}/assets/test.xml'.format(SCRIPT_PATH), 'r') as fileh:
+            cls.test_xml = fileh.read()
+
+    def test_getrunner(self):
+        """Ensure getrunner() can create a runner subclass"""
+        result = runner.getrunner('beaker', {'jobtemplate': 'test'})
+        self.assertIsInstance(result, runner.BeakerRunner)
+
+    def test_invalid_getrunner(self):
+        """Ensure getrunner() throws an exception for an invalid runner"""
+        bad_runner = 'pizza'
+        with self.assertRaises(Exception) as context:
+            runner.getrunner(bad_runner, {'jobtemplate': 'test'})
+
+        self.assertTrue(
+            "Unknown runner type: {}".format(bad_runner)
+            in context.exception
+        )
+
+    def test_getxml(self):
+        """Ensure BeakerRunner.getxml() returns xml"""
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getxml({})
+        self.assertEqual(result, self.test_xml)
+
+    def test_getxml_replace(self):
+        """Ensure BeakerRunner.getxml() returns xml with replacements"""
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getxml({'TEST': 'replaced text'})
+        expected_xml = self.test_xml.replace("##TEST##", "replaced text")
+        self.assertEqual(result, expected_xml)
+
+    @mock.patch('subprocess.Popen')
+    def test_getresultstree(self, mock_popen):
+        """Ensure getresultstree() works"""
+        test_xml = "<xml><test>TEST</test></xml>"
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (test_xml, '')
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getresultstree(jobid=0)
+        self.assertEqual(next(x.text for x in result.iter('test')), 'TEST')
+
+    @mock.patch('subprocess.Popen')
+    def test_dumpjunitresults(self, mock_popen):
```

Comment:
Please document `dumpjunitresults` before testing it. A commit in the same PR would be fine.

Reply:
Done